server/src/controllers/auth.controller.js

```diff
@@ -29,15 +30,15 @@ const signupUser = async (req, res) => {
 
   const newUser = await Users.create({ ...userData });
 
-  const payload = {
-    newUser
-  };
+  const token = cryptoRandomString({ length: 16 });
 
-  const token = generateToken(payload);
+  await EmailVerifications.create({ token, userId: newUser.id });
```

Comment:
why not add an `expiresOn` field to the EmailVerifications model and check it before verifying an email?